src/components/ModelTableList/ModelTable/ModelTable.tsx

```diff
@@ -0,0 +1,208 @@
+        : []),
+      ...(groupBy !== "status"
+        ? [
+            {
+              "data-testid": TestId.COLUMN_STATUS,
+              content: <Status status={highestStatus} />,
+              className: "u-capitalise",
+            },
+          ]
+        : []),
+      ...(groupBy === "cloud"
+        ? [
+            {
+              "data-testid": TestId.COLUMN_REGION,
+              content: (
+                <TruncatedTooltip message={region}>{region}</TruncatedTooltip>
+              ),
+            },
+          ]
+        : [
+            {
+              "data-testid": TestId.COLUMN_CLOUD,
+              content: (
+                <TruncatedTooltip message={generateCloudAndRegion(model)}>
+                  {cloud}
+                </TruncatedTooltip>
+              ),
+            },
+          ]),
+      {
+        "data-testid": TestId.COLUMN_CREDENTIAL,
+        content: (
+          <TruncatedTooltip message={credential}>{credential}</TruncatedTooltip>
+        ),
+      },
+      {
+        "data-testid": TestId.COLUMN_CONTROLLER,
+        content: (
+          <TruncatedTooltip message={controller}>{controller}</TruncatedTooltip>
+        ),
+      },
+      // We're not currently able to get a last-accessed or updated from JAAS.
+      {
+        "data-testid": TestId.COLUMN_UPDATED,
+        content: (
+          <TruncatedTooltip message={lastUpdated}>
+            {lastUpdated}
+          </TruncatedTooltip>
+        ),
+        className: "u-align--right lrg-screen-access-cell",
+      },
+      {
+        "data-testid": TestId.COLUMN_ACTIONS,
+        content: (
+          <ModelActions activeUser={activeUser} modelName={model.model.name} />
+        ),
+      },
+    ];
+
+    const sortData = {
+      name: model.model.name,
+      ...(groupBy !== "owner" ? { owner } : {}),
+      ...(groupBy !== "status" ? { status: highestStatus } : {}),
+      ...(groupBy === "cloud" ? { region } : { cloud }),
+      credential,
+      controller,
+      lastUpdated,
+    };
+
+    const row = {
+      "data-testid": `model-uuid-${model?.uuid}`,
+      columns,
+      sortData,
+    };
+
+    modelsList.push(row);
+  });
+
+  return modelsList;
+}
+
+type Props = {
+  models: ModelData[];
+  groupBy: "status" | "owner" | "cloud";
+  groupLabel: string;
+  emptyStateMessage?: string;
+};
+
+export default function ModelTable({
+  models,
+  groupBy,
+  groupLabel,
+  emptyStateMessage = "",
+}: Props) {
+  const activeUsers = useAppSelector(getActiveUsers);
+  const controllers = useAppSelector(getControllerData);
+  let headerOptions = {};
+  if (groupBy === "status") {
+    headerOptions = {
+      showCloud: true,
+      showOwner: true,
+      ...(groupLabel === "Blocked" ? { showHeaderStatus: true } : {}),
+    };
+  } else if (groupBy === "owner") {
+    headerOptions = { showCloud: true, showStatus: true };
+  } else if (groupBy === "cloud") {
+    headerOptions = { showOwner: true, showStatus: true };
+  }
```

Comment:
Just a small suggestion, if you think it's easier to read:

```suggestion
  const headerOptions = {
    showCloud: ["status", "owner"].includes(groupBy),
    showOwner: ["status", "cloud"].includes(groupBy),
    showStatus: ["owner", "cloud"].includes(groupBy),
      ...(groupLabel === "Blocked" && groupBy === "status" ? { showHeaderStatus: true } : {}),
  };
```

Reply:
Thank you!